pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-inject-parameters.adoc

```diff
@@ -0,0 +1,305 @@
+* `HttpClient`: Enables the policy to make HTTP calls. For more information about how to make HTTP calls from the policy, see xref:policies-pdk-configure-features-http-request.adoc[].
+* `CacheBuilder`: Provides the caching features of the policy. For more information about caching, see xref:policies-pdk-configure-features-caching.adoc[].
+* `StreamProperties`: Provides a structure to share properties with other policies that process the same request. For more information about sharing information between policies, see <<streamproperties>>.
+
+You can also directly inject the `HttpClient` and `StreamProperties` parameter into the `on_request` or `on_response` wrapped functions. For example, if you need to perform an HTTP call inside the `on_request` function, inject `HTTPClient` directly into that function. 
+
+For best practice, inject the parameters only where they must be used. However, if the parameters require initialization steps that need to be executed only one time when the policy is applied, it is best to inject the parameters into the `#[entrypoint]` to complete the configuration steps to increase the performance of the wrapped functions.
+
+== on_request and on_response Wrapped Functions
+
+The wrapped functions accept the following parameters:
+
+* `StreamProperties`
+* `HttpClient`
+* `RequestState`: Provides access to the different stages of request filtering to operate over the incoming requests headers and body. `RequestState` is only available in `on_request` functions. To read and write to the headers and body in the`RequestState`, see xref:policies-pdk-configure-features-headers.adoc[].
+* `ResponseState`: Provides access to the different stages of request filtering to operate over the upstream responses headers and body. `ResponseState` is only available in `on_response` functions. To read and write to the headers and body in the `ResponseState`, see xref:policies-pdk-configure-features-headers.adoc[].
+* `RequestData`: Share data between the `on_request` and`on_response` functions. `RequestData` is only available in `on_response` functions. To learn more about `RequestData`, see xref:policies-pdk-configure-features-share-data.adoc[].
+* `DefaultBindings`: Provides an easy way to bind relevant information to Dataweave evaluators. To learn more about `DefaultBindings`, see xref:policies-pdk-configure-features-dataweave.adoc#defaultbindings[DefaultBindings].
+* `Authentication`: Provides access to read authentication data or share it with other policies. To learn more about `Authentication`, see <<authentication>>.
+
+If your wrapped functions only receive these parameters, the lambda is not required in the wrapper. For example, if the signature of the function is:
+
+[source,Rust]
+----
+async fn request_filter(state: RequestState) -> RequestData<String>;
+async fn response_filter(state: ResponseState, RequestData(path): RequestData<String>);
+----
+
+Instead of a function definition of:
```

Comment:
Does this "instead" relate to the above "If your wrapped functions...", or does it relate to the below "You can use the following...."?